packages/component-library-angular/src/checkbox/Checkbox.spec.ts

```diff
@@ -0,0 +1,365 @@
+    const { fixture } = await render('<input type="checkbox" utrecht-checkbox />', {
+      declarations: [UtrechtCheckboxAttr],
+    });
+
+    expect(fixture.nativeElement).toBeVisible();
+    expect(fixture.nativeElement).toHaveStyle({ display: 'block' });
+  });
+
+  it('renders a design system BEM block class name', async () => {
+    const { fixture } = await render('<input type="checkbox" utrecht-checkbox />', {
+      declarations: [UtrechtCheckboxAttr],
+    });
+    const debugElement = fixture.debugElement.query(By.css('[type="checkbox"]input'))!;
+    expect(debugElement.nativeElement.classList.contains('utrecht-checkbox')).toBeTruthy();
+  });
+
+  describe('checked variant', () => {
+    it('is not checked by default', async () => {
+      const { fixture } = await render('<input type="checkbox" utrecht-checkbox />', {
+        declarations: [UtrechtCheckboxAttr],
+      });
+
+      expect(fixture.nativeElement).not.toBeChecked();
+    });
+
+    it('omits non-essential checked attributes when not checked', async () => {
+      const handleChange = jest.fn();
+      const { fixture } = await render(
+        '<input (ngModelChange)="handleChange()" checked="false" type="checkbox" utrecht-checkbox />',
+        {
+          declarations: [UtrechtCheckboxAttr],
+        },
+      );
+      const debugElement = fixture.debugElement.query(By.css('[type="checkbox"]input'));
+
+      expect(debugElement.nativeElement).not.toHaveAttribute('aria-checked');
+      expect(handleChange).not.toHaveBeenCalled();
+      expect(debugElement.nativeElement).toHaveAttribute('checked');
+    });
+
+    it('can have a checked state', async () => {
+      const handleChange = jest.fn();
+      const { fixture } = await render(
+        '<input (change)="handleChange()" checked="true" type="checkbox" utrecht-checkbox />',
+        {
+          declarations: [UtrechtCheckboxAttr],
+        },
+      );
+      fixture.detectChanges();
+      const debugElement = fixture.debugElement.query(By.css('[type="checkbox"]input'));
+      handleChange();
+
+      expect(handleChange).toHaveBeenCalled();
+      expect(debugElement.nativeElement).toBeChecked();
+    });
+
+    it('can have a checked state in CSS', async () => {
+      const handleChange = jest.fn();
+      const { fixture } = await render(
+        '<input (change)="handleChange()" checked="true" type="checkbox" utrecht-checkbox />',
+        {
+          declarations: [UtrechtCheckboxAttr],
+        },
+      );
+      fixture.detectChanges();
+      const debugElement = fixture.debugElement.query(By.css(':checked'));
+      handleChange();
+
+      expect(handleChange).toHaveBeenCalled();
+
+      expect(debugElement.nativeElement).toBeInTheDocument();
+    });
+  });
+  describe('invalid variant', () => {
+    it('renders a design system BEM modifier class name', async () => {
+      const { fixture } = await render('<input invalid="true" type="checkbox" utrecht-checkbox />', {
```

Comment:
In Angular, for rendering boolean properties, I remember you need to use `[invalid]="true"` instead of `invalid="true"`. The first will be a boolean, because the contents of the attributes are evaluated as JavaScript. The second would be a `"true"` string.